src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs

```diff
@@ -108,6 +112,23 @@ private static bool GetOptionValue(IPluginCommunication autoRest, string option)
             }
         }
 
+        private static string GetOptionStringValue(IPluginCommunication autoRest, string option, Func<string, string>? func)
+        {
+            var projectFolder = autoRest.GetValue<string?>(Options.ProjectFolder).GetAwaiter().GetResult();
+            return projectFolder == null ? GetDefaultOptionStringValue(option)! : (func == null ? projectFolder : func(projectFolder));
+        }
+
+        public static string? GetDefaultOptionStringValue(string option)
```

Comment:
Should the default calculation move into the Configuration.cs? So it's applied to autorest and direct invocation both/

Reply:
I actually did the default calculation in Configuration.ctor in the initial code because of the same reason you pointed out. 

Then I realized other optional options are not doing the same. Configuration.json uses a strategy to only write the value if it's not default. Later when reading the values, it will give the bool option the default value if the option is not found. Putting the write and read code with the handling of default values together makes some sense to me. I used the same strategy for `ProjectFolder` and prefer to handle the options consistently, that's why I made the change.